fili-core/src/main/java/com/yahoo/bard/webservice/sql/helper/CalciteHelper.java

```diff
@@ -0,0 +1,156 @@
+        return new RelToSqlConverter(dialect);
+    }
+
+    /**
+     * Creates a {@link RelBuilder} with a root scema of {@link #DEFAULT_SCHEMA}.
+     *
+     * @param dataSource  The dataSource for the jdbc schema.
+     *
+     * @return the relbuilder from Calcite.
+     *
+     * @throws SQLException if can't readSqlResultSet from database.
+     */
+    public static RelBuilder getBuilder(DataSource dataSource) throws SQLException {
+        return getBuilder(dataSource, DEFAULT_SCHEMA);
+    }
+
+
+    /**
+     * Creates a {@link RelBuilder} with the given schema.
+     *
+     * @param dataSource  The dataSource for the jdbc schema.
+     * @param schemaName  The name of the schema used for the database.
+     *
+     * @return the relbuilder from Calcite.
+     *
+     * @throws SQLException if can't readSqlResultSet from database.
+     */
+    public static RelBuilder getBuilder(DataSource dataSource, String schemaName) throws SQLException {
+        SchemaPlus rootSchema = Frameworks.createRootSchema(true);
+        return RelBuilder.create(
+                Frameworks.newConfigBuilder()
+                        .parserConfig(SqlParser.Config.DEFAULT)
+                        .defaultSchema(addSchema(rootSchema, dataSource, schemaName))
+                        .traitDefs((List<RelTraitDef>) null)
+                        .programs(Programs.heuristicJoinOrder(Programs.RULE_SET, true, 2))
+                        .build()
+        );
+    }
+
+    /**
+     * Adds the schema name to the rootSchema.
+     *
+     * @param rootSchema  The calcite schema for the database.
+     * @param dataSource  The dataSource for the jdbc schema.
+     * @param schemaName  The name of the schema used for the database.
+     *
+     * @return the schema.
+     */
+    private static SchemaPlus addSchema(SchemaPlus rootSchema, DataSource dataSource, String schemaName) {
+        return rootSchema.add(// avg tests run at ~75-100ms
+                schemaName,
+                JdbcSchema.create(rootSchema, null, dataSource, null, null)
+        );
+        // todo look into timing/behavior of cloneschema (above is faster, but it could just be a result of H2)
+        //        rootSchema.setCacheEnabled(true); //almost no effect
+        //        return rootSchema.add( // avg tests run at ~200ms
+        //                schemaName,
+        //                new CloneSchema(
+        //                        rootSchema.add(schemaName, JdbcSchema.create(rootSchema, null, dataSource, null,
+        // null))
+        //                )
+        //        );
+    }
+
+    public SqlPrettyWriter getNewSqlWriter() {
+        return new SqlPrettyWriter(dialect);
+    }
+
+    /**
+     * Gets the name of a table with it's schema (i.e. table to "schema"."table").
+     *
+     * @param sqlTableName  The name of the table to escape.
+     *
+     * @return the escaped schema/table
+     */
+    public String escapeTableName(String sqlTableName) {
```

Comment:
`getTimestampColumn(...)` now takes the schema, and table name separately and unescaped for `DataBaseMetadata::getColumns` so this now just returns `schemaName` 👍